The green add button just dissapeared.

**Description**
The extension was working fine and all of a sudden the green add to anki button disappeared. No idea what broke it. Anki connect is working. 

**Browser version**
Firefox 124.0.2

**Yomitan version**
Yomitan 24.4.7.0
Anki Version ⁨2.1.48

**Exported settings file**
[yomitan-settings-2024-04-14-23-42-13.json](https://github.com/themoeway/yomitan/files/14972555/yomitan-settings-2024-04-14-23-42-13.json)
![image](https://github.com/themoeway/yomitan/assets/44975797/8c802b06-4ca2-4167-942a-74659eedca3d)

Encountered this issue on Kiwi Browser 8265119645, Yomitan 24.4.15.0. I've pasted the console logs below:
```
log.js:138 Yomitan (development build) 24.4.15.0 has encountered a problem.
Originating URL: chrome-extension://ifgadlclliipkebdbniaddamapenidgc/popup.html?type=terms&query=%E7%9C%81%E7%95%A5&wildcards=off
ExtensionError: Anki error: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $
	at com.google.gson.JsonParser.parseReader(JsonParser.java:66)
	at com.google.gson.JsonParser.parseString(JsonParser.java:47)
	at com.kamwithk.ankiconnectandroid.routing.AnkiAPIRouting.findRouteHandleError(AnkiAPIRouting.java:101)
	at com.kamwithk.ankiconnectandroid.routing.APIHandler.chooseAPI(APIHandler.java:33)
	at com.kamwithk.ankiconnectandroid.routing.RouteHandler.get(RouteHandler.java:70)
	at fi.iki.elonen.router.RouterNanoHTTPD$DefaultStreamHandler.post(RouterNanoHTTPD.java:100)
	at fi.iki.elonen.router.RouterNanoHTTPD$UriResource.process(RouterNanoHTTPD.java:386)
	at fi.iki.elonen.router.RouterNanoHTTPD$UriRouter.process(RouterNanoHTTPD.java:596)
	at fi.iki.elonen.router.RouterNanoHTTPD.serve(RouterNanoHTTPD.java:671)
	at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)
	at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $
	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1597)
	at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1404)
	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:542)
	at com.google.gson.stream.JsonReader.peek(JsonReader.java:425)
	at com.google.gson.JsonParser.parseReader(JsonParser.java:61)
	... 11 more

    at AnkiConnect._invoke (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/comm/anki-connect.js:441:31)
    at async AnkiConnect.canAddNotesWithErrorDetail (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/comm/anki-connect.js:138:24)
    at async Backend.detectDuplicateNotes (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:547:16)
    at async Backend.partitionAddibleNotes (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:557:31)
    at async Backend._onApiGetAnkiNoteInfo (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:583:47)
Data: {
    "action": "canAddNotesWithErrorDetail",
    "params": {
        "notes": [
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            }
        ]
    },
    "status": 200,
    "apiError": "com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $\n\tat com.google.gson.JsonParser.parseReader(JsonParser.java:66)\n\tat com.google.gson.JsonParser.parseString(JsonParser.java:47)\n\tat com.kamwithk.ankiconnectandroid.routing.AnkiAPIRouting.findRouteHandleError(AnkiAPIRouting.java:101)\n\tat com.kamwithk.ankiconnectandroid.routing.APIHandler.chooseAPI(APIHandler.java:33)\n\tat com.kamwithk.ankiconnectandroid.routing.RouteHandler.get(RouteHandler.java:70)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$DefaultStreamHandler.post(RouterNanoHTTPD.java:100)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$UriResource.process(RouterNanoHTTPD.java:386)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$UriRouter.process(RouterNanoHTTPD.java:596)\n\tat fi.iki.elonen.router.RouterNanoHTTPD.serve(RouterNanoHTTPD.java:671)\n\tat fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)\n\tat fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)\n\tat java.lang.Thread.run(Thread.java:1012)\nCaused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $\n\tat com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1597)\n\tat com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1404)\n\tat com.google.gson.stream.JsonReader.doPeek(JsonReader.java:542)\n\tat com.google.gson.stream.JsonReader.peek(JsonReader.java:425)\n\tat com.google.gson.JsonParser.parseReader(JsonParser.java:61)\n\t... 11 more\n"
}

Issues can be reported at https://github.com/themoeway/yomitan/issues
```

--- a/ext/js/background/backend.js
+++ b/ext/js/background/backend.js
@@ -537,24 +537,16 @@ export class Backend {
 
     /**
      * @param {import('anki').Note[]} notes
-     * @returns {Promise<({ canAdd: true; } | { canAdd: false; error: string; })[]>}
+     * @returns {Promise<import('backend').CanAddResults>}
      */
-    async detectDuplicateNotes(notes) {
+    async partitionAddibleNotes(notes) {
         // `allowDuplicate` is on for all notes by default, so we temporarily set it to false
         // to check which notes are duplicates.
         const notesNoDuplicatesAllowed = notes.map((note) => ({...note, options: {...note.options, allowDuplicate: false}}));
 
-        return await this._anki.canAddNotesWithErrorDetail(notesNoDuplicatesAllowed);
-    }
-
-    /**
-     * Partitions notes between those that can / cannot be added.
-     * It further sets the `isDuplicate` strings for notes that have a duplicate.
-     * @param {import('anki').Note[]} notes
-     * @returns {Promise<import('backend').CanAddResults>}
-     */
-    async partitionAddibleNotes(notes) {
-        const canAddResults = await this.detectDuplicateNotes(notes);
+        // If only older AnkiConnect available, use `canAddNotes`.
+        const withDuplicatesAllowed = await this._anki.canAddNotes(notes);
+        const noDuplicatesAllowed = await this._anki.canAddNotes(notesNoDuplicatesAllowed);
 
         /** @type {{ note: import('anki').Note, isDuplicate: boolean }[]} */
         const canAddArray = [];
@@ -562,16 +554,11 @@ export class Backend {
         /** @type {import('anki').Note[]} */
         const cannotAddArray = [];
 
-        for (let i = 0; i < canAddResults.length; i++) {
-            const result = canAddResults[i];
-
-            // If the note is a duplicate, the error is "cannot create note because it is a duplicate".
-            if (result.canAdd) {
+        for (let i = 0; i < withDuplicatesAllowed.length; i++) {
+            if (withDuplicatesAllowed[i] === noDuplicatesAllowed[i]) {
                 canAddArray.push({note: notes[i], isDuplicate: false});
-            } else if (result.error.endsWith('duplicate')) {
-                canAddArray.push({note: notes[i], isDuplicate: true});
             } else {
-                cannotAddArray.push(notes[i]);
+                canAddArray.push({note: notes[i], isDuplicate: true});
             }
         }
 

--- a/ext/js/comm/anki-connect.js
+++ b/ext/js/comm/anki-connect.js
@@ -128,15 +128,16 @@ export class AnkiConnect {
         return result;
     }
 
+
     /**
      * @param {import('anki').Note[]} notes
-     * @returns {Promise<({ canAdd: true } | { canAdd: false, error: string })[]>}
+     * @returns {Promise<boolean[]>}
      */
-    async canAddNotesWithErrorDetail(notes) {
+    async canAddNotes(notes) {
         if (!this._enabled) { return []; }
         await this._checkVersion();
-        const result = await this._invoke('canAddNotesWithErrorDetail', {notes});
-        return this._normalizeCanAddNotesWithErrorDetailArray(result, notes.length);
+        const result = await this._invoke('canAddNotes', {notes});
+        return this._normalizeArray(result, notes.length, 'boolean');
     }
 
     /**
@@ -577,52 +578,6 @@ export class AnkiConnect {
             }
         }
         return /** @type {T[]} */ (result);
-    }
-
-    /**
-     * @param {unknown} result
-     * @param {number} expectedCount
-     * @returns {import('anki-connect.js').CanAddResult[]}
-     * @throws {Error}
-     */
-    _normalizeCanAddNotesWithErrorDetailArray(result, expectedCount) {
-        if (!Array.isArray(result)) {
-            throw this._createUnexpectedResultError('array', result, '');
-        }
-        if (expectedCount !== result.length) {
-            throw this._createError(`Unexpected result array size: expected ${expectedCount}, received ${result.length}`, result);
-        }
-        /** @type {import('anki-connect.js').CanAddResult[]} */
-        const result2 = [];
-        for (let i = 0; i < expectedCount; ++i) {
-            const item = /** @type {unknown} */ (result[i]);
-            if (item === null || typeof item !== 'object') {
-                throw this._createError(`Unexpected result type at index ${i}: expected object, received ${this._getTypeName(item)}`, result);
-            }
-
-            const {canAdd, error} = /** @type {{[key: string]: unknown}} */ (item);
-            if (typeof canAdd !== 'boolean') {
-                throw this._createError(`Unexpected result type at index ${i}, field canAdd: expected boolean, received ${this._getTypeName(canAdd)}`, result);
-            }
-
-            if (canAdd && typeof error !== 'undefined') {
-                throw this._createError(`Unexpected result type at index ${i}, field error: expected undefined, received ${this._getTypeName(error)}`, result);
-            }
-            if (!canAdd && typeof error !== 'string') {
-                throw this._createError(`Unexpected result type at index ${i}, field error: expected string, received ${this._getTypeName(error)}`, result);
-            }
-
-            if (canAdd) {
-                result2.push({canAdd});
-            } else if (typeof error === 'string') {
-                const item2 = {
-                    canAdd,
-                    error
-                };
-                result2.push(item2);
-            }
-        }
-        return result2;
     }
 
     /**

--- a/test/playwright/playwright-util.js
+++ b/test/playwright/playwright-util.js
@@ -138,7 +138,7 @@ function getResponseBody(action) {
         case 'deckNames': return ['Mock Deck'];
         case 'modelNames': return ['Mock Model'];
         case 'modelFieldNames': return [...getMockModelFields().keys()];
-        case 'canAddNotesWithErrorDetail': return [{canAdd: true}, {canAdd: true}];
+        case 'canAddNotes': return [true, true];
         case 'storeMediaFile': return 'mock_audio.mp3';
         case 'addNote': return 102312488912;
         case 'multi': return [];

